assets/javascripts/discourse/controllers/group-reports-show.js

```diff
@@ -6,93 +6,103 @@ import Bookmark, {
-          },
-          onAfterDelete: () => {
-            this.set("queryGroup.bookmark", null);
-          },
-        }
-      );
-    },
+  explain = false;
 
-    // This is necessary with glimmer's one way data stream to get the child's
-    // changes of 'params' to bubble up.
-    updateParams(identifier, value) {
-      this.set(`model.params.${identifier}`, value);
-    },
-  }, // actions
+  get hasParams() {
+    return this.model.param_info.length > 0;
+  }
 
-  @discourseComputed("queryGroup.bookmark")
-  bookmarkLabel(bookmark) {
-    return bookmark ? "bookmarked.edit_bookmark" : "bookmarked.title";
-  },
+  get bookmarkLabel() {
+    return this.queryGroupBookmark
+      ? "bookmarked.edit_bookmark"
+      : "bookmarked.title";
+  }
 
-  @discourseComputed("queryGroup.bookmark")
-  bookmarkIcon(bookmark) {
-    if (bookmark && bookmark.reminder_at) {
+  get bookmarkIcon() {
+    if (this.queryGroupBookmark && this.queryGroupBookmark.reminder_at) {
       return WITH_REMINDER_ICON;
     }
     return NO_REMINDER_ICON;
-  },
+  }
 
-  @discourseComputed("queryGroup.bookmark")
-  bookmarkClassName(bookmark) {
-    return bookmark
+  get bookmarkClassName() {
+    return this.queryGroupBookmark
       ? ["query-group-bookmark", "bookmarked"].join(" ")
       : "query-group-bookmark";
-  },
-});
+  }
+
+  @action
+  run() {
+    this.loading = true;
+    this.showResults = false;
+
+    ajax(`/g/${this.get("group.name")}/reports/${this.model.id}/run`, {
+      type: "POST",
+      data: {
+        params: JSON.stringify(this.model.params),
+        explain: this.explain,
+      },
+    })
+      .then((result) => {
+        this.results = result;
+        if (!result.success) {
+          return;
+        }
+
+        this.showResults = true;
+      })
+      .catch((err) => {
+        if (err.jqXHR && err.jqXHR.status === 422 && err.jqXHR.responseJSON) {
+          this.results = err.jqXHR.responseJSON;
+        } else {
+          popupAjaxError(err);
+        }
+      })
+      .finally(() => (this.loading = false));
+  }
+
+  @action
+  toggleBookmark() {
+    return openBookmarkModal(
+      this.queryGroupBookmark ||
+        Bookmark.create({
+          bookmarkable_type: "DataExplorer::QueryGroup",
+          bookmarkable_id: this.queryGroup.id,
+          user_id: this.currentUser.id,
+        }),
+      {
+        onAfterSave: (savedData) => {
+          const bookmark = Bookmark.create(savedData);
+          this.queryGroupBookmark = bookmark;
+          this.appEvents.trigger(
+            "bookmarks:changed",
+            savedData,
+            bookmark.attachedTo()
+          );
+        },
+        onAfterDelete: () => {
+          this.queryGroupBookmark = null;
+        },
+      }
+    );
+  }
+
+  // This is necessary with glimmer's one way data stream to get the child's
+  // changes of 'params' to bubble up.
+  @action
+  updateParams(identifier, value) {
+    this.set(`model.params.${identifier}`, value);
```

Comment:
Small difference, but let's try not to rely on `.set()` in classes that use `@tracked`:

```suggestion
    this.model.set(`params.${identifier}`, value);
```

(this way the underlying `Controller` class could drop EmberObject dependency and this will still work)

Reply:
added in https://github.com/discourse/discourse-data-explorer/pull/218/commits/69690b905c345a4b6105ea9766392a3239ecbad8